### ISSUE TITLE
Alt text on ggplot2 graphs should be carried forward into HTML docs produced by Quarto

### ISSUE THREAD

**Issue by BajczA475** [quarto-dev/quarto-r#307]
This is a feature request. In ggplot2, you can now set the alt text of graphs via ggplot2::labs(alt = ...). However, this alt text is not then automatically applied to a graph embedded in a knitted Quarto document, and this seems like a significant missed opportunity. I have no idea what the mechanical barriers to this might be, but it seems like if ggplot2::get_alt_text(PLOT) produces an output, it could be fed automatically to the fig-alt attribute.

**Comment by cderv** [comment]
Thank for the feature request. 

I believe this is something to be handled at a lower level that the quarto R package. To my knowledge this is tracked already in 
- https://github.com/yihui/knitr/issues/2001

Then it will work on R Markdown and Quarto. 

Does that cover ? 

> I have no idea what the mechanical barriers to this might be, but it seems like if ggplot2::get_alt_text(PLOT) produces an output, it could be fed automatically to the fig-alt attribute.

yes you have good intuition. You should already be able to do this 
````
```{r}
#| fig-alt: !expr last_alt
library(ggplot2)

p <- ggplot(mtcars, aes(x = wt, y = mpg)) +
  geom_point() +
  labs(alt = "example alt text")
last_alt <- ggplot2::get_alt_text(p)
p
```
````

**Comment by BajczA475** [comment]
Cool! Do we know where that knitr Issue currently stands? It's nice that there's a somewhat intuitive pass-through already, but considering that many are still loath to include alt test at all, I think the easier it can be to set the alt text once and be done with it, the more likely it will be that folks will actually do it 😅.

**Comment by cderv** [comment]
> Do we know where that knitr Issue currently stands?

We need to create the mechanism. Leave a 👍 there to show interest. We'll get to it. 

And I'll close here so we'll follow there.
